docs/plugins/building_a_plugin/_layer_data_guide.md

```diff
@@ -6,6 +6,11 @@ directly. Instead, it passes (mostly) pure-python and array-like types,
 deconstructed into a {class}`tuple` that we refer to as a `LayerData` tuple.  This type shows
 up often in plugins and is explained here.
 
+Note that when writing your own plugin, type annotations are optional, 
+except in the case of [`magicgui` function widgets](magicgui).
+Here we sometimes provide a type annotation with a more informative name, 
+indicating its value's _functional role_ in a plugin, in addition to its possible types.
```

Comment:
You were right in that they are type aliases! Sorry :facepalm: 
So I think all npe2 type aliases are defined [here](https://github.com/napari/npe2/blob/main/src/npe2/types.py).

So maybe something like:
```suggestion
Napari defines several type aliases for `LayerData`  related types. These provide more
informative names that indicates the value's _functional role_ in a plugin.
We describe them below.
```

technically its npe2 not napari but I think this is okay.

Reply:
I use a wording very similar to what you suggest, here: https://github.com/napari/docs/pull/367/commits/a8a053eb99b6e85c263611f09f94eb710546ca8f